core/models/tool_configuration.rb

```diff
@@ -30,10 +30,19 @@ def save
       file.write(YAML.dump(@config))
       file.close
       config_file = File.expand_path(CONFIG_FILE_NAME, XDG['CONFIG_HOME'].to_s)
+      check_config_dir(config_file)
+
       FileUtils.cp(file.path, config_file)
     end
   end
 
+  # Checks the config directory and creates a directory if it is missing
+  def check_config_dir(config_file)
+    config_dir = File.dirname(config_file)
+    out = FileUtils.mkdir_p(config_dir)
+    raise "Cannot create directory for configuration file: #{out}" unless out.first == config_dir
```

Comment:
If you raise the exception, then it must be processed in the method that initiated the call of this method. Just raising the exception is not sufficient.

Please also explain when this exception can be even raised? I do not get the logic here.